src/api/policystore-client.js

```diff
@@ -0,0 +1,50 @@
+/*
+ * DECODE App – A mobile app to control your personal data
+ *
+ * Copyright (C) 2019 – DRIBIA Data Research S.L.
+ *
+ * DECODE App is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * DECODE App is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * email: info@dribia.com
+ */
+
+import axios from 'axios';
+import { debugLog } from 'lib/utils';
+import PolicystoreError from './errors/policystore-error';
+
+const prefix = '/twirp/decode.iot.policystore.PolicyStore';
+
+class PolicystoreClient {
+  constructor(url = 'https://policystore.decodeproject.eu') {
+    this.url = url;
+  }
+
+  async listPolicies() {
+    try {
+      const url = `${this.url}${prefix}/ListEntitlementPolicies`;
+      debugLog('Going to call: ', url);
+
+      const { policies } = await axios.post(url, {}, { headers: { 'Content-Type': 'application/json' } });
```

Comment:
need to understand what happens when axios gets back an error http response - does this go down the error handling route, or do we need to handle that in this clause?

Reply:
It depends on what you consider an ‘error http response’ which depends on how the service works. 
If the serviec returns an error HTTP status axios will throw an error, and the client must construct a suitable PolicystoreError in the catch block. 
If it returns a 2xx but there are some conditions in the data returned that make the client decide the response is not good, then the client can throw the PolicystoreError itself after checking for that condition in the try block.